ocaml/libs/tracing/tracing.ml

```diff
@@ -132,14 +134,15 @@ module Span = struct
     ; attributes
     }
 
-  let get_tag t tag = snd (List.find (fun s -> fst s = tag) t.attributes)
+  let get_tag t tag = Attributes.find tag t.attributes
 
   let finish ?(attributes = []) ~span () =
-    {
-      span with
-      end_time= Some (Unix.gettimeofday ())
-    ; attributes= span.attributes @ attributes
-    }
+    let attributes =
+      List.fold_left
+        (fun attrs (k, v) -> Attributes.add k v attrs)
+        span.attributes attributes
+    in
```

Comment:
I'd rather use union here to be explicit on which map do we prefer here when merging the two maps:
```suggestion
    (* prefer attributes from the parameter *)
    let attributes = Attributes.union
      (fun _k a _b -> Some a)
      attributes span.attributes
    in
```
https://v2.ocaml.org/api/Map.S.html#VALunion